### ISSUE TITLE
2.0.0a2 Comment in a complex, nested sub-query breaks indentation of following SQL

### ISSUE THREAD

**Issue by dmohns** [sqlfluff/sqlfluff#4252]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Took me a while to figure out what the issue here was, probably there is a simpler MWE, but I was only able reproduce it in this specific scenario. 

Sometimes a comment can cause problems with following indentation. Consider the following example, everything after the comment has 1 indent too much

```sql
WITH cte AS (
    SELECT *
    FROM (
        SELECT *
        FROM table
        WHERE
            NOT bool_column AND NOT bool_column
            AND some_column >= 1  -- This is a comment
        )
    ),

    SELECT *
    FROM cte
;

    SELECT *
    FROM table3
;
```

However, if I remove the comment everything gets linted correctly

```sql
WITH cte AS (
    SELECT *
    FROM (
        SELECT *
        FROM table
        WHERE
            NOT bool_column AND NOT bool_column
            AND some_column >= 1
    )
),

SELECT *
FROM cte
;

SELECT *
FROM table3
;
```

### Expected Behaviour

see above

### Observed Behaviour

see above

### How to reproduce

see above

### Dialect

BigQuery

### Version

```
sqlfluff, version 2.0.0a2
```

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
